src/frontends/tensorflow/src/op/assign.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_op_table.hpp"
+#include "openvino/frontend/tensorflow/node_context.hpp"
+#include "openvino/frontend/tensorflow/variable.hpp"
+
+using namespace std;
+using namespace ov;
+using namespace ov::op;
+using namespace ov::frontend::tensorflow;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+OutputVector translate_assign_op(const NodeContext& node) {
+    default_op_checks(node, 2, {"Assign"});
+    auto ref = as_type_ptr<Variable>(node.get_input_by_reference(0).get_node_shared_ptr());
+    TENSORFLOW_OP_VALIDATION(
+        node,
+        ref,
+        "[TensorFlow Frontend] internal error: Assign operation expects variable by the first input");
+
+    // create new variable operation node that will represent the same variable but it will be initialized with value
+    auto value = node.get_input(1);
+    auto new_ref = make_shared<Variable>(*ref, value);
```

Comment:
Do we need `set_node_name()` for this op?

Reply:
No, we do not need it because the operation outputs reference and not regular tensor. I would not set either tensor name nor node name for such operations to avoid confusion.